pkg/controllers/dnspolicy/dnspolicy_controller.go

```diff
@@ -93,8 +93,11 @@ func (r *DNSPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		if !markedForDeletion {
 			if apierrors.IsNotFound(err) {
 				log.V(3).Info("Network object not found. Cleaning up")
-				err := r.deleteResources(ctx, dnsPolicy, nil)
-				return ctrl.Result{}, err
+				delResErr := r.deleteResources(ctx, dnsPolicy, nil)
+				if delResErr == nil {
+					delResErr = err
+				}
+				return r.reconcileStatus(ctx, dnsPolicy, fmt.Errorf("%w : %w", conditions.ErrTargetNotFound, delResErr))
```

Comment:
Similar to how kuadrant operator currently deals with reconciling the status https://github.com/Kuadrant/kuadrant-operator/blob/main/controllers/authpolicy_controller.go#L66-L70

Reply:
would it work to return the specErr here at the end? I think we want it reported but also want it requeued? Other option would be to wrap the notfound err in a targetNotFoundErr and then have the reconcile status return a requeue on specific error types?

Reply:
Yeah I think that is the best option for now. 
1) return an error or requeue
2) follow on to requeue policies (perhaps based on some label constructed based on the target ref) if an addEvent happens

Reply:
https://github.com/Kuadrant/multicluster-gateway-controller/issues/610